support/oidc-discovery-provider/handler.go

```diff
@@ -34,14 +44,31 @@ func (h *Handler) serveWellKnown(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if !h.domains[r.Host] {
+		http.Error(w, "domain not allowed", http.StatusNotFound)
+		return
+	}
+
+	if h.allowInsecureScheme {
```

Comment:
this only serves the urls as HTTP if allowInsecureScheme is configured and if the origin request is not served on HTTPS. So in allowInsecure is enabled but the Reverse Proxy runs on HTTPS, it is still served on HTTPS.